prophic3/array_axiom_enumerator.h

```diff
@@ -16,50 +16,33 @@ class ArrayAxiomEnumerator {
       : ts_(ts), abstractor_(aa) {
     msat_env_ = ts.get_env();
 
-    TermTypeMap & orig_types = abstractor_.orig_types();
-
     // sort the indices
     // convenient to store them grouped by current and all for 1-step and 2-step
     // lemmas
-    std::string typestr;
     msat_term base_idx; // gets assigned the actual var if it's wrapped in ubv_to_int
     for (auto idx : abstractor_.indices()) {
-      typestr = msat_type_repr(abstractor_.get_orig_type(idx));
-
-      // HACK keep original type over current and next state version
-      orig_types[ts.cur(idx)] = orig_types.at(idx);
-      orig_types[ts.next(idx)] = orig_types.at(idx);
 
       // save state variable indices
       if (ts.only_cur(idx))
       {
-        state_indices_[typestr].insert(idx);
-        all_indices_[typestr].insert(ts.next(idx));
+        state_indices_.insert(idx);
+        all_indices_.insert(ts.next(idx));
       }
 
       if (!ts.contains_next(idx))
       {
-        curr_indices_[typestr].insert(idx);
-        curr_indices_no_witnesses_[typestr].insert(idx);
+        curr_indices_.insert(idx);
+        curr_indices_no_witnesses_.insert(idx);
       }
-      orig_indices_[typestr].insert(idx);
+      orig_indices_.insert(idx);
```

Comment:
what is the difference between orig_indices_ and orig_indices_set_?

Reply:
Good catch! There's no difference anymore. Changed with: 2b4ad50fb7796987b1570987ab179c988496b3cd

By the way, I think I'll also get rid of most of these different index sets and instead detect the indices on the fly like you've mentioned before. I'll make that in a separate PR.

Reply:
cool.